AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/cryptographic-materials-cache.smithy

```diff
@@ -210,12 +210,20 @@ structure StormTrackingCache {
   sleepMilli: CountingNumber,
 }
 
+@documentation("Shared cache across multiple Hierarchical Keyrings.")
+structure SharedCache {
+  @required
+  @documentation("Shared cache across multiple Hierarchical Keyrings.")
+  cache: CryptographicMaterialsCacheReference,
+}
+
 union CacheType {
   Default : DefaultCache,
   No: NoCache,
   SingleThreaded: SingleThreadedCache,
   MultiThreaded: MultiThreadedCache,
-  StormTracking: StormTrackingCache
+  StormTracking: StormTrackingCache,
+  Shared: SharedCache
```

Comment:
I think you can remove the indirection and just put the reference here? 

Or did you try that and run into a problem?

Reply:
Yeah I was just trying out something. Updated Shared to be ICMC directly without the additional structure `sharedCache`.